fork-management-for-infra-providers/README.md

```diff
@@ -0,0 +1,148 @@
+- We sometimes need to maintain some diffs in the core code since
+  - We need hotfixes and we don't have time to wait for merges in upstream repositories.
+  - We have a pretty specific temporary/permanent requirement and it is not possible to add it to upstream repositories. We must fight for keeping this diff as small/short-term as possible. 
+
+We need to 
+- define a fork management policy including repo management and branching strategies
+- automate code syncs, image building and releases 
+
+## Current status
+
+### Repositories
+
+We have 2 repositories per operator
+1. Fork of the upstream repository 
+- Examples:
+  - https://github.com/giantswarm/cluster-api
+  - https://github.com/giantswarm/cluster-api-provider-aws
+  - https://github.com/giantswarm/cluster-api-provider-cloud-director
+  - https://github.com/giantswarm/cluster-api-provider-gcp
+
+1. A repository for bundling the operator as an app 
+- Naming: https://github.com/giantswarm/\<fork-name\>-app
+- Examples
+  - https://github.com/giantswarm/cluster-api-app
+  - https://github.com/giantswarm/cluster-api-provider-aws-app
+  - https://github.com/giantswarm/cluster-api-provider-cloud-director-app
+  - https://github.com/giantswarm/cluster-api-provider-gcp-app
+
+### Syncing fork repositories
+
+There is no common way.
+
+In CAPVCD, since we have some diffs between our fork and upstream repo, we have a mechanism as follows:
+- `main` branch of the repo is the same as the upstream one. We sync this branch by clicking `Sync fork` button in the UI.
+- `giantswarm-main` branch of the repo contains our changes. We manually rebase it on top of `main` branch when necessary and open a new PR each time. The developer is supposed to resolve conflicts during rebasing.
+ 
+### Bundling + Customization
+
+There are some differences between repositories but the general structure in app repositories is as follows:
+
+- We have Makefile + Bash scripts to
+  - Check out a specific commit/branch of fork/upstream repo. [Example](https://github.com/giantswarm/cluster-api-provider-cloud-director-app/blob/adfc74115d9b7aeda60d885500122be3fb434d2e/Makefile.custom.mk#L14)
+  - Generate base manifests by running scripts of the core repo
+  - Customize base manifests with kustomization. [Example](https://github.com/giantswarm/cluster-api-provider-cloud-director-app/blob/adfc74115d9b7aeda60d885500122be3fb434d2e/Makefile.custom.mk#L19)
+  - Update manifests under `helm` folder for the app. [Example](https://github.com/giantswarm/cluster-api-provider-cloud-director-app/blob/adfc74115d9b7aeda60d885500122be3fb434d2e/Makefile.custom.mk#L28)
+
+When it is necessary to update the app content, developers are supposed to
+- run make targets manually to update manifests and values in the app repo
+- open a PR in the app repo
+
+### Image Management
+
+For some providers, we retag upstream images with retagger.
+For some providers like CAPVCD, we build/push images manually.
```

Comment:
Why is CAPVCD different?

Reply:
Since we have some diffs between our fork and the upstream repo.